File: 프로그래머스/2/12913. 땅따먹기/땅따먹기.swift
import Foundation

func solution(_ land:[[Int]]) -> Int{
    var answer = 0
    var dp = land
    
    for i in 1..<land.count{
        dp[i][0] += max(dp[i-1][1],max(dp[i-1][2],dp[i-1][3]))
        dp[i][1] += max(dp[i-1][0],max(dp[i-1][2],dp[i-1][3]))
        dp[i][2] += max(dp[i-1][0],max(dp[i-1][1],dp[i-1][3]))
        dp[i][3] += max(dp[i-1][0],max(dp[i-1][1],dp[i-1][2]))
    }
        
    
    
    
    return dp[land.count-1].max()!
}

File: 프로그래머스/2/12913. 땅따먹기/땅따먹기.py
from copy import deepcopy

def solution(land):
    l = len(land)
    for i in range(1,l):
        for j in range(0,4) :
                land[i][j] += max(land[i-1][:j]+land[i-1][j+1:])
        
    return max(land[l-1])